src/metabase/upload.clj

```diff
@@ -317,11 +316,12 @@
         column-names       (keys col->upload-type)]
     (driver/create-table! driver db-id table-name col->database-type)
     (try
-      (let [rows (parsed-rows col->upload-type csv-file)]
-        (driver/insert-into! driver db-id table-name column-names rows)
-        {:num-rows    (count rows)
-         :num-columns (count column-names)
-         :size-mb     (/ (.length csv-file) 1048576.0)})
+      (with-open [reader (io/reader csv-file)]
```

Comment:
key change: keeping the file open while we do the insert, so that we can insert line by line.